### PR TITLE
Code refactor

### PR DESCRIPTION
**Pull request by DimaNevelev**

### CODE REVIEW

File: artifactory-tasks-utils/genericTaskExecutor.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            tl.rmRF(specPath);
+        } catch (ex) {
+            taskRes = "Failed cleaning temporary FileSpec file.";
+            tl.setResult(tl.TaskResult.Failed, taskRes);
+        }
+
+        if (taskRes) {
+            return;
+        }
+        tl.setResult(tl.TaskResult.Succeeded, "Build Succeeded.");
```
Comment (yahavi) [inline artifactory-tasks-utils/genericTaskExecutor.js]: You can simplify in all task to:
```
if (!taskRes) {
    tl.setResult(tl.TaskResult.Succeeded, "Build Succeeded.");
}
```

File: tasks/ArtifactoryNuget/nugetBuild.js [hunk 2]
```diff
@@ -133,16 +88,51 @@ function addNugetArgsToCommands() {
...
+}
+
+function execPushCommand(command, cliPath, buildDir) {
+    let targetDeployRepo = tl.getInput("targetDeployRepo");
+    let pathToNupkg = utils.fixWindowsPaths(tl.getPathInput("pathToNupkg", true, false));
+    command.addCommand(cliUploadCommand).addArguments(pathToNupkg, targetDeployRepo);
+    runNuGet(command, cliPath, buildDir);
+}
+
+if (utils.isWindows()) {
```
Comment (eyalbe4) [inline tasks/ArtifactoryNuget/nugetBuild.js]: This looks like a bug - should probably be if (!utils.isWindows()) {

File: tests/tests.js [hunk 3]
```diff
@@ -12,6 +12,7 @@ let tasksOutput;
```
Comment (eyalbe4) [inline tests/tests.js]: Some tests have been added to to this file by other commits.
Please merge them.

**Comment by DimaNevelev** [comment]
@eyalbe4,
Please see my [comment](https://github.com/jfrog/artifactory-vsts-extension/pull/35#discussion_r226868897) and changes.

**Pull request closed without merging**